### PR TITLE
mention map_blocks in the docstring of apply_ufunc

### LINKED ISSUE #5010: DataArrays inside apply_ufunc with dask=parallelized
**Is your feature request related to a problem? Please describe.**
Currently, when using apply_ufunc with `dask=parallelized` the wrapped function receives numpy arrays upon computation.  
Some xarray operations generate enormous amount of chunks (best example : `da.groupby('time.dayofyear')`, so any complex script using dask ends up with huge task graphs. Dask's scheduler becomes overloaded,  sometimes even hangs, sometimes uses way more RAM than its workers.

**Describe the solution you'd like**
I'd want to profit from both the tools of xarray and the power of dask parallelization. I'd like to be able to do something like this:

```python3
def func(da):
     """Example of an operation not (easily) possible with numpy."""
     return da.groupby('time').mean()

xr.apply_ufunc(
    da,
    func,
    input_core_dims=[['time']],
    pass_xr=True,
    dask='parallelized'
)
```
I'd like the wrapped func to receive DataArrays resembling the inputs (named dims, coords and all), but only with the subset of that dask chunk.  Doing this, the whole function gets parallelized : dask only sees 1 task and I can code using xarray. Depending on the implementation, it might be less efficient than `dask=allowed` for small dataset, but I think this could be beneficial for long and complex computations on large datasets.

**Describe alternatives you've considered**
The alternative is to reduce the size of the datasets (looping on other dimensions), but that defeats the purpose of dask.

Another alternative I am currently testing, is to add a layer between apply_ufunc and the `func`. That layer reconstruct a DataArray and deconstructs it before returning the result, so xarray/dask only passing by. If this works and is elegant enough, I can maybe suggest an implementation within xarray.

### PR DESCRIPTION
**Pull request by keewis**
Should we also add a `Notes` section like the one in `apply_ufunc`?

- [x] Closes #5010
- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

**Comment by dcherian** [comment]
yes lets copy over the ntoes

File: xarray/core/computation.py [hunk 1]
```diff
@@ -923,6 +923,12 @@ def apply_ufunc(
     Single value or tuple of Dataset, DataArray, Variable, dask.array.Array or
     numpy.ndarray, the first type on that list to appear on an input.
 
+    Notes
+    -----
+    This function is designed for the more common case where ``func`` can work on numpy
+    arrays. If ``func`` needs to manipulate a whole xarray object subset to each block
+    it is possible to use ``map_blocks``.
```
Comment (keewis) [inline xarray/core/computation.py:930]: do we need to state that it is recommended to use `apply_ufunc` wherever possible?

Comment (dcherian) [reply xarray/core/computation.py:930]: sure.

**Comment by dcherian** [comment]
Thanks @keewis

**Pull request merged**